### PR TITLE
[CopyFilesOverSSH] Migrated from scp2 to ssh2-sftp-client

### PR DESCRIPTION
**Pull request by sdobrodeev**
**Task name**: CopyFilesOverSSH

**Description**: Migration from currently unmaintained scp2 library to ssh2-sftp-client

**Documentation changes required:** (Y/N) No

**Added unit tests:** (Y/N) No

**Attached related issue:** (Y/N) https://github.com/microsoft/azure-pipelines-tasks/issues/13048

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [ ] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/CopyFilesOverSSHV0/sshhelper.ts [hunk 1]
```diff
@@ -159,41 +140,41 @@ export class SshHelper {
...
-                if(code && code != 0) {
+                if (code && code != 0) {
                     //non zero exit code - fail
                     defer.reject(tl.loc('RemoteCmdNonZeroExitCode', cmdToRun, code));
                 } else {
                     //no exit code or exit code of 0
 
                     //based on the options decide whether to fail the build or not if data was written to STDERR
-                    if(stdErrWritten === true && options.failOnStdErr === true) {
+                    if (stdErrWritten === true && options.failOnStdErr === true) {
```
Comment (anton-erofeev) [inline Tasks/CopyFilesOverSSHV0/sshhelper.ts:177]:
```suggestion
                    if (stdErrWritten && options.failOnStdErr) {
```

File: Tasks/CopyFilesOverSSHV0/sshhelper.ts [hunk 2]
```diff
@@ -159,41 +140,41 @@ export class SshHelper {
      * @param options
      * @returns {Promise<string>}
      */
-    runCommandOnRemoteMachine(command: string, options: RemoteCommandOptions) : Q.Promise<string> {
+    runCommandOnRemoteMachine(command: string, options: RemoteCommandOptions): Q.Promise<string> {
         var defer = Q.defer<string>();
-        var stdErrWritten:boolean = false;
+        var stdErrWritten: boolean = false;
```
Comment (anton-erofeev) [inline Tasks/CopyFilesOverSSHV0/sshhelper.ts:145]: Can we replace `var` with `let` here?

File: Tasks/CopyFilesOverSSHV0/sshhelper.ts [hunk 3]
```diff
@@ -159,41 +140,41 @@ export class SshHelper {
...
 
         this.sshClient.exec(cmdToRun, (err, stream) => {
-            if(err) {
+            if (err) {
                 defer.reject(tl.loc('RemoteCmdExecutionErr', cmdToRun, err))
             }
             stream.on('close', (code, signal) => {
                 tl.debug('code = ' + code + ', signal = ' + signal);
-                if(code && code != 0) {
+                if (code && code != 0) {
```
Comment (AlexandrKravchuk) [inline Tasks/CopyFilesOverSSHV0/sshhelper.ts:170]:
```suggestion
                if (code && code !== 0) {
```

File: Tasks/CopyFilesOverSSHV0/sshhelper.ts [hunk 4]
```diff
@@ -159,41 +140,41 @@ export class SshHelper {
...
-        if(!options) {
+        if (!options) {
             tl.debug('Options not passed to runCommandOnRemoteMachine, setting defaults.');
             var options = new RemoteCommandOptions();
             options.failOnStdErr = true;
         }
 
         var cmdToRun = command;
-        if(cmdToRun.indexOf(';') > 0) {
+        if (cmdToRun.indexOf(';') > 0) {
```
Comment (AlexandrKravchuk) [inline Tasks/CopyFilesOverSSHV0/sshhelper.ts:158]:
```suggestion
        if (cmdToRun.find(';')) {
```

**Comment by damccorm** [comment]
Can we add units or canary tests for this? Changing dependencies like this is fine, but makes me nervous with no test coverage

**Comment by sdobrodeev** [comment]
> Can we add units or canary tests for this? Changing dependencies like this is fine, but makes me nervous with no test coverage

Canary tests passed -https://dev.azure.com/buildcanary/PipelineCanary/_build/results?buildId=944253&view=logs&j=377bee81-b530-5eab-ac40-709ee4d08646&t=377bee81-b530-5eab-ac40-709ee4d08646

**Comment by oddstr13** [comment]
This may have caused a regression in that directories are no longer automatically created
jellyfin/jellyfin#3553
I assume this is why it's being reverted?

**Comment by anatolybolshakov** [comment]
Hi @oddstr13, yes, it was reverted since it caused regression with destination directories no longer created

**Pull request merged**